lib/routes.js

```diff
@@ -40,24 +40,38 @@ export function isHealthy() {
     return true;
 }
 
-export function setHealthCheckResponse(req, res, log) {
+function writeResponse(res, error, log) {
+    const statusCode = error ? error.code : 200;
+    const body = error ? JSON.stringify(error) : '{}';
+
+    res.writeHead(statusCode, { 'Content-Type': 'application/json' });
+    res.write(body);
+
+    res.end(() => {
+        log.end().info('healthcheck ended', {
+            httpCode: res.statusCode,
+        });
+    });
+}
+
+function healthcheckRouteHandler(req, res, log) {
     if (isHealthy()) {
         if (req.method === 'GET' || req.method === 'POST') {
-            res.writeHead(200, { 'Content-Type': 'application/json' });
-            res.write('{}');
-            log.debug('healthcheck response',
-                { httpMethod: req.method, httpCode: res.statusCode });
+            writeResponse(res, null, log);
```

Comment:
Not related to the MPU error but I just realized that writeResponse is async and is not completed before the return of this method. This may trick the test into believing that the operation is complete while its not.